src/config.rs

```diff
@@ -86,3 +93,33 @@ pub enum OperatorFormat {
     Short = 1,
     Numeric = 2,
 }
+
+#[derive(Debug, Clone)]
+pub enum Apn {
+    Given(String<100>),
+    Automatic,
```

Comment:
My idea behind the `Automatic option`, was to add something like https://gitlab.gnome.org/GNOME/mobile-broadband-provider-info or similar, as a LUT, to do automatic APN resolution

Reply:
I like the idea of that but maybe putting it behind a feature flag as it would require substantial amount of flash.
Will maybe also move user and pass into the enum.

Reply:
That makes sense :+1: 
And completely agree on the feature flag. Also not something you need to implement now, it was just to give some context on the enum :)

Reply:
I also don't think the Given needs to be 100 chars :p Not sure what the actual limit is, but i would assume much lower

Reply:
I found in different documents different max length for the APN. Some 100 some 64, most probably 64 is enough for everything.